numba/core/config.py

```diff
@@ -119,6 +119,25 @@ def update(self, force=False):
             # Store a copy
             self.old_environ = dict(new_environ)
 
+        self.validate()
+
+    def validate(self):
+        global CUDA_USE_NVIDIA_BINDING
+
+        if CUDA_USE_NVIDIA_BINDING:  # noqa: F821
+            try:
+                import cuda  # noqa: F401
+            except ImportError as ie:
+                msg = ("CUDA Python bindings requested (the environment "
+                       "variable NUMBA_CUDA_USE_NVIDIA_BINDING is set). "
+                       f"but they are not importable: {ie.msg}")
+                warnings.warn(msg)
+
+                CUDA_USE_NVIDIA_BINDING = False
+
+            if CUDA_PER_THREAD_DEFAULT_STREAM:  # noqa: F821
```

Comment:
Now that `CUDA_USE_NVIDIA_BINDING` is a request only, should this condition now also require `CUDA_USE_NVIDIA_BINDING` to be set?

Reply:
I think if it was initially set, the warning should always appear, even if it turn out that the request could not be honoured - otherwise the user is in danger of thinking that the requested configuration was valid apart from not being able to find the bindings.

Reply:
That's a good point, let's leave as is.